src/Middleware/ForwardedFor.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Amp\Http\Server\Middleware;
+
+use Amp\Socket\InternetAddress;
+
+final class ForwardedFor
+{
+    /**
+     * @param array<non-empty-string, string|null> $fields
+     */
+    public function __construct(
+        private readonly InternetAddress $address,
+        private readonly array $fields,
+    ) {
+
+    }
+
+    public function getAddress(): InternetAddress
+    {
+        return $this->address;
+    }
+
+    public function getFieldValue(string $key): ?string
+    {
+        return $this->fields[$key] ?? null;
+    }
```

Comment:
The header is well specified, no? So we can add proper getters?

Reply:
Yes, I should have mentioned I intended to expand upon this object. I was being a bit lazy for the PoC :wink:

Reply:
Updated with proper getters.